### PR TITLE
Update to support protobuf 3.0.

### LINKED ISSUE #1821: Support protobuf 3.0 when released.
@JakeWharton 
Hi Wharton，
Retrofit is really a delightful networking framework for android. But we have recently found “ProtoConverterFactory” is not support ProtoBuf 3.0. So we want to know when you can update the ProtoConverter.

Thank you very much.

### PR DESCRIPTION
**Pull request by JakeWharton**
Closes #1821.

### CODE REVIEW

File: retrofit-converters/protobuf/src/main/java/retrofit2/converter/protobuf/ProtoConverterFactory.java [hunk 1]
```diff
@@ -61,12 +63,20 @@ private ProtoConverterFactory(ExtensionRegistryLite registry) {
...
     try {
-      Field field = c.getDeclaredField("PARSER");
+      Method method = c.getDeclaredMethod("parser");
       //noinspection unchecked
-      parser = (Parser<MessageLite>) field.get(null);
-    } catch (NoSuchFieldException | IllegalAccessException e) {
-      throw new IllegalArgumentException(
-          "Found a protobuf message but " + c.getName() + " had no PARSER field.");
+      parser = (Parser<MessageLite>) method.invoke(null);
+    } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException ignored) {
```
Comment (swankjesse) [inline retrofit-converters/protobuf/src/main/java/retrofit2/converter/protobuf/ProtoConverterFactory.java]: pedantic: if it’s an InvocationTargetException, falling back doesn’t seem right

**Comment by swankjesse** [comment]
LGTM! Nice trick supporting two versions simultaneously.

**Comment by JiongBull** [comment]
Thanks, i'm waiting for it, when it can be used for proto3.

**Comment by JakeWharton** [comment]
There's nothing preventing the current converter from being used with
proto3.

**Comment by JiongBull** [comment]
@JakeWharton  sincere thanks

File: retrofit-converters/protobuf/src/main/java/retrofit2/converter/protobuf/ProtoConverterFactory.java (hunk 1 above)
Comment (JakeWharton) [inline retrofit-converters/protobuf/src/main/java/retrofit2/converter/protobuf/ProtoConverterFactory.java]: Rethrow the cause? It should never happen.

**Comment by fabriciointerama** [comment]
@JakeWharton  Hi. I'm new with java and android dev. That said, sorry if I say some silly stuff here. I was trying your updated factory class but compiler says this error for your [multi-catch](https://github.com/square/retrofit/blob/f7a8ba0b448a8643bdb8e989f19468d6c365e8ee/retrofit-converters/protobuf/src/main/java/retrofit2/converter/protobuf/ProtoConverterFactory.java#L71): 

> **Multi-catch** with these reflection exceptions requires API level 19 (current min is 16) because they get compiled to the common but new super type ReflectiveOperationException. As a workaround either create individual catch statements, or catch Exception.

And since [Retrofit says that](https://github.com/square/retrofit):

> Retrofit requires at minimum Java 7 or **Android 2.3**.

I don't know if this is an issue, so I thought it would be better to note it.

**Comment by JakeWharton** [comment]
Thanks I'll have a look.

**Comment by JakeWharton** [comment]
This was determined to not be an issue in the library. We actually have a compile-time check in place which prevents this from happening.

**Pull request merged**